### PR TITLE
add support for %{shrink:}

### PR DESCRIPTION
**Pull request by ignatenkobrain**

### CODE REVIEW

File: rpmio/macro.c [hunk 1]
```diff
@@ -864,6 +865,27 @@ doFoo(MacroBuf mb, int negate, const char * f, size_t fn,
 	if ((b = strrchr(buf, '/')) != NULL)
 	    *b = '\0';
 	b = buf;
+    } else if (STREQ("shrink", f, fn)) {
+	/*
+	 * shrink body by removing all leading and trailing whitespaces and
+	 * reducing intermediate whitespaces to a single space character.
+	 */
+	int i, j, k, was_space = 0;
+	for (i = 0, j = 0, k = (int)strlen(buf); i < k; ) {
```
Comment (pmatilai) [inline rpmio/macro.c]: Use the appropriate type for the variable so you dont need a silly cast like this.

File: rpmio/macro.c [hunk 2]
```diff
@@ -864,6 +865,27 @@ doFoo(MacroBuf mb, int negate, const char * f, size_t fn,
...
+	    if (isspace((int)(buf[i]))) {
```
Comment (pmatilai) [inline rpmio/macro.c]: Drop the redundant cast. Also you can use risspace() and drop the <ctype.h> include from the top after that.

File: rpmio/macro.c [hunk 3]
```diff
@@ -864,6 +865,27 @@ doFoo(MacroBuf mb, int negate, const char * f, size_t fn,
...
+		was_space = 1;
+		i++;
+		continue;
+	    } else if (was_space) {
+		was_space = 0;
+		if (j > 0) /* remove leading blanks at all */
+		    buf[j++] = ' ';
+		/* fallthrough */
```
Comment (pmatilai) [inline rpmio/macro.c]: The /* fallthrough */ comment seems out of place, there's no switch-case here. A leftover porting artifact maybe?

**Comment by ignatenkobrain** [comment]
@pmatilai I think I fixed all nitpicks ;)

**Comment by pmatilai** [comment]
Merged manually (commit 9a9eefc40e56821a2bc12de78114c623a7d5a461 and 7fe183d66008c379bcdeb8fd82a8585924347b92), I added back the mention it's ported from rpm5.

**Pull request closed without merging**